File: fundamentos/desafio-js/src/questions/how-many-acc-w-name-maria-at-ag-47.ts
import { IAccount } from "../i-account";

/** Quantos clientes que tem Maria no nome estão na agencia 47? */
export default function howManyAccWNameMariaAtAg47(...accounts: Array<IAccount>) {
  const sum = accounts
    .filter(acc => acc.agencia == 47 && acc.name.match(/maria/gi))
    .length;

  console.info(`Quantos clientes que tem Maria no nome estão na agencia 47? ${sum}`);
}

File: fundamentos/desafio-dart/lib/questions/how_many_acc_w_name_maria_at_ag_47.dart
import 'package:desafio_dart/domains/account.dart';

/// Quantos clientes que tem Maria no nome estão na agencia 47?
void howManyAccWNameMariaAtAg47(List<Account> accounts) {
  final sum = accounts
      .where((acc) =>
          acc.agencia == 47 &&
          acc.name
              .contains(RegExp('maria', multiLine: true, caseSensitive: false)))
      .length;

  print('Quantos clientes que tem Maria no nome estão na agencia 47? ${sum}');
}
